diff -Nr -U3 pkg_makefiles.orig/bison_host.mk pkg_makefiles/bison_host.mk
--- pkg_makefiles.orig/bison_host.mk	2020-03-21 14:54:58.233177276 +0800
+++ pkg_makefiles/bison_host.mk	2020-03-21 23:35:44.948148747 +0800
@@ -23,4 +23,5 @@
 
 ${DIR_HOST}:
 	tar -zxf ${STORAGE}/${DIR}.tar.gz
+	cd ${DIR} && patch -p0 < ${OCTEON_ROOT}/gnu-patches/xxxx-bison-2.6.5-io-eof.patch
 	mv ${DIR} ${DIR_HOST}
diff -Nr -U3 pkg_makefiles.orig/flex_host.mk pkg_makefiles/flex_host.mk
--- pkg_makefiles.orig/flex_host.mk	2020-03-21 15:10:20.315446732 +0800
+++ pkg_makefiles/flex_host.mk	2020-03-21 23:36:26.271413314 +0800
@@ -23,5 +23,5 @@
 
 ${DIR_HOST}:
 	tar -zxf ${STORAGE}/${DIR}.tar.gz
+	cd ${DIR} && patch -p1 < ${OCTEON_ROOT}/gnu-patches/xxxx-flex-2.6.4.patch
 	mv ${DIR} ${DIR_HOST}
-
diff -Nr -U3 pkg_makefiles.orig/flex.mk pkg_makefiles/flex.mk
--- pkg_makefiles.orig/flex.mk	2020-03-21 15:10:03.935248100 +0800
+++ pkg_makefiles/flex.mk	2020-03-21 23:36:07.419738723 +0800
@@ -21,4 +21,5 @@
 
 ${DIR}:
 	tar -zxf ${STORAGE}/${PKG}-${VERSION}.tar.gz 
+	cd ${DIR} && patch -p1 < ${OCTEON_ROOT}/gnu-patches/xxxx-flex-2.6.4.patch 
 
diff -Nr -U3 pkg_makefiles.orig/m4_host.mk pkg_makefiles/m4_host.mk
--- pkg_makefiles.orig/m4_host.mk	2020-03-21 15:09:41.742980691 +0800
+++ pkg_makefiles/m4_host.mk	2020-03-21 23:36:53.218976330 +0800
@@ -3,7 +3,7 @@
 include ../.config
 BUILD:=${shell pwd}
 PKG:=m4
-VERSION:=1.4.16
+VERSION:=1.4.18
 DIR:=${PKG}-${VERSION}
 DIR_HOST:=${PKG}_host-${VERSION}
 
@@ -23,5 +23,6 @@
 
 ${DIR_HOST}:
 	tar -zxf ${STORAGE}/${DIR}.tar.gz
+	cd ${DIR} && patch -p0 < ${OCTEON_ROOT}/gnu-patches/xxxx-m4-1.4.18-io-eof.patch
 	mv ${DIR} ${DIR_HOST}
 
